### PR TITLE
Social: Add social post toggle component

### PR DESCRIPTION
**Pull request by gmjuhasz**
This change adds the new component which has the checkbox for  toggling `shouldUploadAttachedMedia` that was introduced in #30179

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

1204272127368434-as-1204411047112874
## Does this pull request change what data or activity we track or use?

No.
## Testing instructions:

* Have the Advanced plan on your site, and check out this PR.
* Open a new post. In the Jetpack/Social tab there should be a new checkbox in the Media section. Toggle it on, then save post as draft. Value of the checkbox should remain after refresh.

CleanShot 2023-04-19 at 14 51 05 png

### CODE REVIEW

File: projects/js-packages/publicize-components/src/components/social-post-control/index.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+					checked={ shouldUploadAttachedMedia }
+					onChange={ onCheckboxChange }
+					label={ __( 'Share as a social post', 'jetpack' ) }
+				/>
+				<Text variant="small" className={ styles.description }>
+					{ __(
+						'By default, the image and text appear on social media as a link preview. To optimize engagement, share as a social post.',
+						'jetpack'
+					) }
+				</Text>
```
Comment (danielpost) [inline projects/js-packages/publicize-components/src/components/social-post-control/index.js]: Is it possible to use the `help` attribute on `CheckboxControl` for this instead?

File: projects/js-packages/publicize-components/src/components/social-post-control/styles.module.scss [hunk 2]
```diff
@@ -0,0 +1,9 @@
+.container {
+	margin-bottom: 20px;
+	// padding-right: 8px;
```
Comment (danielpost) [inline projects/js-packages/publicize-components/src/components/social-post-control/styles.module.scss]: I think this snuck in :)

File: projects/js-packages/publicize-components/src/components/social-post-control/styles.module.scss [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+}
+
+.description.description {
+	line-height: normal;
```
Comment (danielpost) [inline projects/js-packages/publicize-components/src/components/social-post-control/styles.module.scss]: TIL you can use `line-height` with a text value.

File: projects/js-packages/publicize-components/src/components/social-post-control/index.js (hunk 1 above)
Comment (gmjuhasz) [reply projects/js-packages/publicize-components/src/components/social-post-control/index.js]: I missed that it has a `help` attribute 😬 Thanks.

File: projects/js-packages/publicize-components/src/components/social-post-control/styles.module.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.container {
+	margin-bottom: 20px;
```
Comment (danielpost) [inline projects/js-packages/publicize-components/src/components/social-post-control/styles.module.scss:2]: Is it possible to add this to the checkbox itself and get rid of the container?

**Review by danielpost — APPROVED** [review]
LGTM!

**Pull request merged**